homeassistant/components/tplink/README.md

```diff
@@ -17,15 +17,18 @@ use modules.
 
 ## Dynamic feature creation
 
-Entities can be dynamically created by inspecting various properties of a feature such
-as `type`, `unit` or `precision`. If more data is needed than the feature provides,
-i.e. to set HA `device_class` or `state_class` on a sensor entity, then a static `EntityExtras` entry
-should be created in `const.py`.
+Entity attributes are dynamically set by inspecting various properties of a feature such
+as `type`, `unit` or `precision`.
+If more data is needed than the feature provides,
+i.e. to set HA `device_class` or `state_class` on a sensor entity, then add it to the static
+`PLATFORM_DESCRIPTIONS` entry in the appropriate platform.
+All feature ids should be described in the static entity descriptions but if a new feature
+is not yet added it will be created manually and a warning will be logged.
 
 ### Translation keys and icons
 
-For `Feature` based platforms translation keys should match `feature.id`.
-If a translation key and icon has been added to `strings.json` and `icons.json` it should be added to
-`ENTITY_EXTRAS` as an indication not to set the `name` or `icon` from the feature,
-as it will override the translation otherwise.
+For features to use translation keys they should be added to `strings.json` and `icons.json`
+with the feature.id as key.
+
+**All described features must have corresponding entries in `strings.json` and `icons.json`**
```

Comment:
Would you mind already adding these for the existing descriptions? I can extend it with the ones I'm seeing after we get this merged to the `tplink_sensor_rewrite` branch.

Reply:
I think we have all the `strings` in place for existing features. w.r.t `icons` I think we have all the ones we need that are not provided by the base platforms? https://github.com/home-assistant/core/pull/120064 updates readme with a note about translations provided by HA.